src/pgduckdb_utils.cpp

```diff
@@ -50,7 +44,7 @@ CreateOrGetDirectoryPath(const char* directory_name) {
 	const auto duckdb_data_directory = oss.str();
 
 	if (!CheckDirectory(duckdb_data_directory)) {
-		if (MakePGDirectory(duckdb_data_directory.c_str()) == -1) {
+		if (mkdir(duckdb_data_directory.c_str(), S_IRWXU) == -1) {
```

Comment:
Should use the `pg_dir_create_mode` instead. That global is only set during startup, so accessing it is fine even when using multiple threads. A Postgres database directory can be configured such that the group also has permissions:

```
/*
 * Set create modes and mask to use when writing to PGDATA based on the data
 * directory mode passed.  If group read/execute are present in the mode, then
 * create modes and mask will be relaxed to allow group read/execute on all
 * newly created files and directories.
 */
void
SetDataDirectoryCreatePerm(int dataDirMode)
```